datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/TestKafkaConnectorTask.java

```diff
@@ -172,6 +172,9 @@ public void testConsumeWithStartingOffset() throws Exception {
 
     KafkaConnectorTask connectorTask = createKafkaConnectorTask(task);
 
+    // validate auto.offset.reset config is overridden to none (given the start offsets)
+    Assert.assertEquals(connectorTask.getConsumerAutoOffsetResetConfig(), "none");
```

Comment:
Can we add a specific test case to try the use-cases out? Many cases crop up since it is 2 configs, but may be we need at least the one you are fixing, with start position specified and no reset strategy.

Reply:
+1, I'd go as far as to make sure that we see the expected behavior when start positions are present, and when they aren't.

Reply:
testConsumeBaseCase tests when no startOffsets are present, 
testConsumeWithTheStartingOffset tests the case with startOffsets are present. It was testing with no auto.offset.rest set, I changed it to set auto.offset.reset set to earliest, and it already verifies the expected behavior (in terms of number of records consumed)

The actual value of the `auto.offset.reset` is irrelevant, we only need to test if it is set that we do override. Whether it was `earliest`,  `latest` or `none`, is orthogonal.

Reply:
I added a test case for when the auto offset reset strategy is set in metadata, to verify startOffsets takes precedence